src/components/common/TmSessionExplore.vue

```diff
@@ -170,18 +178,24 @@ export default {
         return false
       }
     },
-    isANetworkAddress(param) {
-      const selectedNetwork = this.addressPrefixes.find(
-        ({ id }) => id === this.network
+    async selectNetworkByAddress(address) {
+      let selectedNetwork = this.addressPrefixes.filter(({ address_prefix }) =>
+        address.startsWith(address_prefix)
       )
       // handling query not loaded yet or failed
-      if (!selectedNetwork) return false
-
-      if (param.startsWith(selectedNetwork.address_prefix)) {
-        return true
+      if (!selectedNetwork) {
+        console.error("Connecting to the selected network failed")
```

Comment:
This should throw to prevent the process to continue, right?

Reply:
Yes, this is what the next line, the `return`, does

Reply:
ah right! but it never displays an error to the user, or I don't see it.

Reply:
ah no, the return will not prevent succeeding of line 145